### DISCUSSION TITLE
Keeping host process alive

### DISCUSSION THREAD

**Discussion by edfenergy-richardcannock** [quickfix-j/quickfixj#453] [Q&A]
In order to keep an initiator alive within a main process, what is the recommended method? 

In our initiator application, we are connecting to an external exchange, and are required to send an period heatbeat message to indicate the application is still running and the application should be durable and run 24/7 ideally.

Blocking the main thread after starting the initiator with an infinite while loop ala:

initiator.start();

while(true)
{
   thread.sleep(1000); // or whatever really.
}

 initiator.stop();

seems inelegant, but i'm not sure of how else to implement this in a production application.

without the while loop, the application just exits. I assume this is because the initator app is somehow launched on a background thread?

This may be more of a question around generic Java process programming per se but seems relevant in quickfix j application so asking here.

Any advice gratefully received. Thanks.

**Comment by QwertGold** [comment]
Normally, when you start an initiator, your start a thread, that will be a reconnect loop (`ReconnectInterval`) , when the acceptor sends back the first message, a 'session' is establish  (technically the session is created based on the schedule). 
In general you never turn off your message loop, but keep receiving messages and processing them.

**Reply by QwertGold** [reply]
We have a Spring Boot application. In our setup the start of the initiator is linked to the startup on the Spring Context, as long as the application is alive the Initiator is running. When the Spring boot application is shutdown, the initiator terminates. As always, you need to manage the threads of your application (unless this is done for you by a framework).
